### ISSUE TITLE
Symbolでコンソーシアムチェーン（プライベートチェーン）を立てる部

### ISSUE THREAD

**Issue by bootarou** [ymuichiro/symbol_japan_forum#13]
ブロックチェーン界隈ではスケールアップのためにサイドチェーン、サブチェーンなどのL2と最近では異なるチェーン間での取引であるlayerZero（L0）がある。　https://news.yahoo.co.jp/articles/01c36a897b07809ece319db039c0924de2b1df99
Symbolでもアトミックスワップが内蔵されていてプライベートチェーン、コンソーシアムチェーンなどSymbolを用いた別チェーンとも相互に取引が可能となっている。
今後ブロックチェーンの発展と共にL0のような複数チェーンのインターオペラビリティは必ず必要になっていくと考えます。
Ｓｙｍｂｏｌを用いたコンソーシアムチェーンはDHPで実働しているがそのノードの設定方法など私も知らないことが多いのでみなさんと一緒に勉強していきたいと思います！！

**Comment by 0x070696E65** [comment]
あ、これはただのテストネットですかね、、、

**Comment by ymuichiro** [comment]
ちなみに手法ではないので話はそれますが、そのL0の利用シーンとして私が適切だと考えているゴール案置いておきます
https://mobility-transformation.com/magazine/cityos/

勿論もっと小規模でもいいですが

**Comment by bootarou** [comment]
あっ私も都市OSとか大好物ですｗ

https://symbol.github.io/symbol-sdk-typescript-javascript/0.23.3/enums/_model_network_networktype_.networktype.html
ネットワークタイプはいくつかあるようです。プライべートもあるのでこれを使うのかもです。
↓DHPのネットワークプロパティです。
http://vmi780631.contaboserver.net:3000/network/properties　
↓通常のメインネットのネットワークプロパティ。
http://sym-main-01.opening-line.jp:3000/network/properties

ネットワークタイプは同じ。。。謎
違いはエポックタイムとgenerationhashと見えます。
あとcurrencyMosaicIdですね。
ハッシュ適当にセットしたら行けてしまうのかな？ｗ

設定の説明はこちらに。
https://qiita.com/daoka/items/f8c32574a2e223b226c3
と
https://qiita.com/salaryman-toushi/items/a1c399bee5505b250836

**Comment by 0x070696E65** [comment]
ちなみにこれ、僕が立てたものですがつまりこれはすでに新たなブロックチェーンになるんですかね？

https://toshinet.hideyoshi-node.net:3001/network/properties

**Comment by ishidad2** [comment]
こんにちは。
多分プライベートネットワークの立ち上げはこちらが参考になるかもしれません。
まだ私も全然わかってないのですが、ネメシスの設定値をいじっていくみたいです。

https://github.com/symbol/symbol/blob/dev/client/catapult/docs/RUNNETWORKLIN.md

**Comment by bootarou** [comment]
Toshiさんのやつエポックタイムとジェネハッシュと言うことでは新たなチェーンですよね！
daiさんのリンクの内容で設定されたのですか？

でもこの後どうしたらいいのか分からないｗ
同じ設定で立ち上げれば同期し始めるのかな？

**Comment by 0x070696E65** [comment]
僕の場合は
https://docs.symbol.dev/ja/guides/network/creating-a-private-test-net.html
https://github.com/fboucquez/symbol-bootstrap
このへんを参考に
`symbol-bootstrap start -p bootstrap -a multinode -c customPreset.yml`
をやっただけなんです。
ただ、Discordのxemboolさんのコメントを見るに、これはやはりテストネット構築のためのコマンドなのかなと。これだけでも新たなチェーンにはなっててブロック高も上がってるし、色々カスタムはできてアグリゲート1000個内包とかで遊んでました。
ただし、単体ノードでしかなく、他のノードの接続が分からないですｗ

で、どっかで他のノード接続の話見たなーと思って見直したら
https://dev.to/fboucquez/how-to-create-a-symbol-testnet-network-4hdd
ここの最下部で
Allow new nodes to connect:
If you want another person to connect to your network, share with him the custom-network-preset.yml and nemesis-seed folder. Then this person can run:
とあるので、この記事を参考にすればいいのかな？とも思いますが自分がやってないのでなんとも言えませんｗ
記事の内容を見る限りこれが新たなネットワーク立ち上げの方法な気はします！

daiさんのはやってないので全然わからないっす！

**Comment by ymuichiro** [comment]
なんかDiscordで h_gocchiさんがコンソーシアムチェーン作る話をしている？

スペース nemJapanUserGroup
チャンネル 雑談
↓
実はSymbolを使ったコンソーシアムチェーンを作る構想がいくつかありまして、bootstrapのいまの方向性は厄介ですね・・・

**Comment by ishidad2** [comment]
ごめんなさい、実は自分で提示しておいてなんですが、リンク先の内容はよくわかっていません😅
ただ、リンク先はマニュアルビルド後のbinフォルダ以下に出来るバイナリを使用するので手軽にはできないかもです。。


https://github.com/fboucquez/symbol-bootstrap/blob/dev/src/commands/config.ts#L29
一応ここで、my-custom-network.yml指定しての起動が出来るみたいなんで、netwok.ymlがあればイケそうな気はしますけど。そもそもmy-custom-network.ymlの内容がよくわからないｗ

最近はラズパイでのマニュアルビルドばっかりやってたのでsymbol-bootstrapの方は全くわかんないですｗ
ただ、Discordで h_gocchiさんやXEMBookさんが言われているようにbootstrapの今の実装ではコンソーシアムチェーンを作るのは難しいのかも・・（もなかさんがなんか面白そうなことやっておられるみたですけどねｗ）

**Comment by 0x070696E65** [comment]
こちらの記事を参考にプライベートメインネットワークを建ててみました
https://dev.to/fboucquez/how-to-create-a-symbol-testnet-network-4hdd

やることはここに書かれてあるとおりでした。
ただ、僕がもう一つのノードを建てた方法が最初に対話式で作成されるノードのうちの１つを違うサーバーでunzipしそれを run -d するだけで立ち上がるという簡易的な方法で、他のピアノードはここ `https://github.com/0x070696E65/symcomnet` にある `custom-network-preset.yml` と `nemesis-seed` を共有し `symbol-bootstrap start -p custom-network-preset.yml -a dual  -c my-new-node-custom-preset.yml ` とのことですので、もしどなたかご興味があれば。

同時に蛇口とエクスプローラーを作成できました。
蛇口はAPIノードがオフラインとのことで動いてませんが、おそらく参照している場所が違うとかそんなものな気がしますが詳しく調べてません。そのためモザイクの配布が僕からのみになっちゃいますがｗ

- main node http://main.symcom.club:3000/network/properties
- dual node https://dual-001.symcom.club:3001/node/info
- faucet http://main.symcom.club:100/
- explorer http://main.symcom.club:90/

**Comment by ymuichiro** [comment]
若干スレチかもしれませんが、ご意見もらいたく。
都市OSのモックみたいなものを作りたいと思っています。
このコンソーシアムチェーンを立てるの一貫の中で適切なアカウント管理の方法も考えたいなぁと思ってます

でまずは一番基本的なアカウントの生成と管理・認証の所だけで考えているのですが、
これまでのWEBサービスの認証って ID/PW --> OAuth --> MFA や 認証局/NW分離等色々進んできている中で
以下が課題としてあると思っていて

- コスト高い & 複雑
- 他のプラットフォームとの互換性がない（アカウント）
- サーバーが集中砲火を浴びるので常に気が抜けない&運用コスト高い

いっそシンプルにこんな感じの仕組みでシンプルに認証する仕組みを持ったシステムを作ったらいいかなぁと思ったのですが微妙ですかね
（ サーバーは基本求められたキーのデータは返す、ただしクライアントサイドで復号しないと読めない）

https://docs.google.com/presentation/d/1PxNj6NoXvyvnDJk2T8yDgJVG2Od10gwmuCSbva4NE2A/edit#slide=id.gf3e3c79f81_0_11

うまくいったら K8s に コンソーシアムなSymbolとこれらのシステムを搭載してPoC用として撒きたいなと

**Comment by bootarou** [comment]
プライベートチェーンは立てれるみたいですねー！
ふぁーさんのはSSLをSymbolウォレットで行うイメージですか？
私も極論言えば自分の秘密鍵をもって暗号化インターネットできればお面白いと思うのでアリだと思います！

**Comment by ymuichiro** [comment]
> ふぁーさんのはSSLをSymbolウォレットで行うイメージですか？
そうです。そんな感じです。最初から全部暗号化しておければより攻撃者のコストを上げ、Internetの平和に繋がるのではという感じです。
＋ DID も出来ればいいなぁという感じ

> お面白いと思うのでアリだと思います！
ありがとうございます！というわけで作りながら考えてみます！！

**Comment by matsuno3** [comment]
[https://twitter.com/monakaJP/status/1512477375371841537](https://twitter.com/monakaJP/status/1512477375371841537)
Catapultのマニュアルビルド
これは資料になるかな？
